### DISCUSSION TITLE
When clicking link while on a 404 page, the page is reloaded

### DISCUSSION THREAD

**Discussion by cpuix** [inertiajs/inertia#559] [Help]
### Versions:

- `@inertiajs/inertia` version: "^0.8.4"
- `@inertiajs/inertia-vue` version: "^0.5.5"

I have such a problem in the Inertia and Laravel project. When I go to the 404 page and click on any routable menu, the page refreshes and sends. What could be the reason for this?

**Comment by reinink** [comment]
I think I know what the issue here is.

Are you setting up a custom 404 error page via your Laravel exception handler?

The issue here is that the Inertia config is happening in the middleware, which is not run when hitting a 404. This config includes your asset version.

More details here: https://github.com/inertiajs/inertia-laravel/issues/176

Our long term solution here is to remove the middleware, and switch to a service provider approach to setting up Inertia.

However, in the meantime, you can get around this by manually setting your asset version in your `AppServiceProvider` by calling `Inertia::version()`.

```php
use Inertia\Inertia;

Inertia::version($version);
Inertia::version(fn () => $version); // Lazily
```
